tests/ui/test_ui_tools.py

```diff
@@ -1536,19 +1536,26 @@ def test_private_message_to_self(self, mocker):
         ('<a href="http://foo">http://foo</a>',
          [('msg_link', 'http://foo'), ' ', ('msg_link_index', '[1]')]),
         ('<a href="http://foo/bar.png">http://foo/bar.png</a>',
-         [('msg_link', 'http://foo/bar.png'), ' ', ('msg_link_index', '[1]')]),
+         [('msg_link', 'bar.png'), ' ', ('msg_link_index', '[1]')]),
         ('<a href="http://foo">bar</a>',
          [('msg_link', 'bar'), ' ', ('msg_link_index', '[1]')]),
         ('<a href="/user_uploads/blah"',
-         [('msg_link', '{}/user_uploads/blah'.format(SERVER_URL)), ' ',
-          ('msg_link_index', '[1]')]),
+         [('msg_link', 'blah'), ' ', ('msg_link_index', '[1]')]),
         ('<a href="/api"',
-         [('msg_link', '{}/api'.format(SERVER_URL)), ' ',
-          ('msg_link_index', '[1]')]),
+         [('msg_link', 'api'), ' ', ('msg_link_index', '[1]')]),
         ('<a href="some/relative_url">{}/some/relative_url</a>'
          .format(SERVER_URL),
-         [('msg_link', '{}/some/relative_url'.format(SERVER_URL)), ' ',
-          ('msg_link_index', '[1]')]),
+         [('msg_link', 'relative_url'), ' ', ('msg_link_index', '[1]')]),
+        ('<a href="http://foo.com/bar">foo.com/bar</a>',
+         [('msg_link', 'bar'), ' ', ('msg_link_index', '[1]')]),
+        ('<a href="http://foo.com">foo.com</a>'
+         '<a href="http://foo.com">http://foo.com</a>'
+         '<a href="https://foo.com">https://foo.com</a>'
+         '<a href="http://foo.com">Text</a>',
+         [('msg_link', 'http://foo.com'), ' ', ('msg_link_index', '[1]'),
```

Comment:
This is my only remaining query. It's minor really, but if you look at #**commits** then you'll see some on a `chat.zulip.org` branch - it gets shown as `http://chat.zulip.org`, which of course isn't the branch name :)